src/cuda_eval.cpp

```diff
@@ -430,8 +433,17 @@ static void jitc_cuda_render_var(uint32_t index, const Variable *v) {
             break;
 
         case VarKind::Sqrt:
-            fmt(jitc_is_single(v) ? "    sqrt.approx.ftz.$t $v, $v;\n"
-                                  : "    sqrt.rn.$t $v, $v;\n", v, v, a0);
+            if (jitc_is_half(v))
+                fmt("    cvt.f32.f16 %f0, $v;\n"
```

Comment:
Given that this pattern repeats in a few places, how about we have a special per-instruction flag stating whether such a conversion is needed? Then those two lines could be added before/after the switch statement. And we can temporarily overwrite the `v->reg_index` so that `$v` reaches the right operand.

 We'll need to reuse this approach if we ever add 8-bit arithmetic, etc.

Reply:
Makes sense, that change has been added!